tests/base/src/test/java/org/wildfly/security/http/client/ElytronHttpClientTest.java

```diff
@@ -0,0 +1,124 @@
+
+        AuthenticationContext context = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-http-client-basic.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        context.run(() -> {
+            try {
+                HttpServerAuthenticationMechanism mechanism = basicFactory.createAuthenticationMechanism("BASIC", Collections.emptyMap(),getCallbackHandler("user1", "test-realm", "password1"));
+                HttpRequest request = elytronHttpClient.getRequest("http://localhost:8080/servlet-security/SecuredServlet");
+
+                //Test successful authentication
+                TestingHttpServerRequest testingHttpServerRequest = new TestingHttpServerRequest(new String[]{request.headers().allValues("Authorization").get(0)});
+                mechanism.evaluateRequest(testingHttpServerRequest);
+                Assert.assertEquals(Status.COMPLETE,testingHttpServerRequest.getResult());
+            }catch (Exception e){
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    public void testRequest2() throws Exception{
+        AuthenticationContextConfigurationClient AUTH_CONTEXT_CLIENT =
+                doPrivileged((PrivilegedAction<AuthenticationContextConfigurationClient>) AuthenticationContextConfigurationClient::new);
+
+        AuthenticationContext context = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-http-client-basic.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        context.run(() -> {
+            try {
+                Map<String, Object> props = new HashMap<>();
+                props.put(CONFIG_REALM, "RealmUsersRoles");
+                props.put("org.wildfly.security.http.validate-digest-uri", "false");
+                HttpServerAuthenticationMechanism mechanism = digestFactory.createAuthenticationMechanism("DIGEST", props,getCallbackHandler("user1", "RealmUsersRoles", "password1"));
+                TestingHttpServerRequest request1 = new TestingHttpServerRequest(null);
+                mechanism.evaluateRequest(request1);
+                TestingHttpServerResponse response = request1.getResponse();
+                HttpRequest request2 = elytronHttpClient.getResponseHeader(response.getAuthenticateHeader());
+                System.out.println(request2.headers());
```

Comment:
Just a minor, we should not use system.out.println statements in the resulting code